storage/cloudspanner/map_storage_test.go

```diff
@@ -42,12 +58,68 @@ func GetTestDB(ctx context.Context, t *testing.T) *spanner.Client {
 	return client
 }
 
+// Unique per test binary invocation
+var timestamp = time.Now().UTC().Format("jan-02-15-04-05")
+var testBinary = strings.ToLower(strings.Replace(path.Base(os.Args[0]), ".test", "", 1))
+var invocationID = fmt.Sprintf("%s-%s", timestamp, testBinary)
+var dbCount uint32 // Unique per test invocation
+
+func uniqueDBName(project, instance string) string {
+	database := fmt.Sprintf("%s-%d", invocationID, atomic.AddUint32(&dbCount, 1))
```

Comment:
Did you consider using `sync.Mutex` rather than `atomic`?

It'll be an extra line or two, but I think it'll be more clear why it's there;`atomic` is pretty low level and, e.g., I could see someone extending this function later and referencing `dbCount` directly without using `LoadInt32` - would that be a bug/introduce potential for undefined behaviour?

I don't really know for sure without going and doing some reading on memory barriers and stuff, and there are probably others out there like me :)

Reply:
I started off with sync.Mutex, but other team members thought that atomic was better.

Reply:
Switched to sync.Mutex

Reply:
Cool, thanks - I think that's the right approach.

FWIW, here's [Ian Lance Taylor talking on atomic vs. mutexes](https://groups.google.com/d/msg/golang-nuts/0uPDCRiBWqc/X11KP_IaocYJ):
>The behaviour of mutexes is guaranteed by the Go memory model
(http://golang.org/ref/mem).  The behaviour of the sync/atomic
functions is not.
>
>Perhaps the memory model should discuss the sync/atomic functions.  A
downside is that that would just encourage people to use them, which
is almost certainly a bad idea.  Another downside is that I don't
think it's possible to describe the sync/atomic functions in terms of
a happens-before relationship.  I think you have to resort to odd
circumlocutions like "if a LoadInt32 happens to follow a StoreInt32 to
the same memory address, then anything that happens before the
StoreInt32 also happens before the LoadInt32."  But there is no way to
say whether the LoadInt32 follows the StoreInt32 or not, and if it
does not come later then you know nothing about the state of memory.